Домашнее задание

xafish opened the pull request:
hw01
Введение в Spring Framework

xafish committed:
Домашнее задание
hw01
Введение в Spring Framework

xafish committed:
Домашнее задание
hw01
Введение в Spring Framework
Добавлено модульное тестирование

kataus commented on hw01/src/main/java/ru/otus/spring/QuestionModule.java:
```diff
@@ -0,0 +1,35 @@
+        ApplicationContext applicationContext = new ClassPathXmlApplicationContext("spring-context.xml");
+        var questionsDao = applicationContext.getBean(QuestionsDao.class);
+
+        Scanner in = new Scanner(System.in);
```
Вся логика должны быть под контекстом. Снаружи только инициализация контекста и запуск 1 метода 1 бина

kataus commented on hw01/src/main/java/ru/otus/spring/dao/QuestionsDaoImpl.java:
```diff
@@ -0,0 +1,34 @@
+
+    public QuestionsDaoImpl(String filePath) {
+        questions = new HashMap<>();
+        var fileManager = new FileManager(filePath);
```
Конструкторы для функциональных классов не надо вызывать вручную. У нас DI, эту работу сделает  Spring

xafish committed:
Домашнее задание
hw01
Введение в Spring Framework
Добавлено модульное тестирование
+Добавил di, перенёс логику QuestionModule в dao и убрал логику из main

kataus commented on hw01/src/main/java/ru/otus/spring/dao/QuestionsDaoImpl.java:
```diff
@@ -0,0 +1,53 @@
+    public QuestionsDaoImpl(FileManager fileManager) {
+        questions = new HashMap<>();
+
+        List<String[]> csvArray = fileManager.getCsvArray();
```
Конструктор - не лучшее место для тяжелого функционала. Если хочется закешировать - то лучше специализированный бин сделать и лениво туда загружать

kataus commented on hw01/src/main/java/ru/otus/spring/dao/QuestionsDaoImpl.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    @Override
+    public void runTest(){
```
В QuestionDao совсем не ожидается runTest. DAO расшифровывается как Data Access Object.

kataus commented on hw01/src/main/java/ru/otus/spring/utils/FileManager.java:
```diff
@@ -0,0 +1,36 @@
+import java.util.List;
+
+@AllArgsConstructor
+public class FileManager {
```
Интерфейсы - очень сильный инструмент первичной абстракции кода. Их активное использование очень удобно и реально помогает строить гибкий код. Не стоит ими пренебрегать
